conda/recipes/conda_build_config.yaml

```diff
@@ -22,7 +22,7 @@ boost:
   - 1.77
 
 numpy:
-  - 1.19
+  - 1.24.*
```

Comment:
Could I make a suggestion that we match `conda-forge` here unless there is anything that we need from a much more up to date verison? `numpy` is core to so many projects and we use a large set of libraries from conda-forge so having our minimum requires be the same gives us maximum compatibility.

Conda Forge maintain a [pinning repository](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml) where they define these versions.

For numpy they define a version per Python version which is accomplished by matching the list entries for [Python](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L656) and [numpy](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L595), e.g `3.8` uses `1.20`.